docs/add_data_source.md

```diff
@@ -0,0 +1,151 @@
+# How-To Add a Data Provider
+
+When adding a new data provider and input structure, there are several areas of the system
+that must be updated. The system [settings.yml](../config/settings.yml) file must be updated
+to include your new provider in the harvester jobs. A [configuration file](#config) must be written, and
+optionally a [traject macro](#macro).
+
+## Steps
+
+1. [Add source to](#settings) [settings.yml](../config/settings.yml)
+2. [Establish data mapping](#mapping)
+3. [Write spec to test data mapping and provide fixture data for testing](#test)
+4. [Write mapping configuration](#config)
+5. [[Optional] Write mapping macro for data transformations](#macro)
+
+### Settings <a name="settings" />
+
+1. At the end of the sources section, add your new source.
+2. Give it a meaningful id (see existing examples): `my_new_source`
+3. Assign the directory within the repository that data files will be located (see: [dlme-metadata](https://github.com/waynegraham/dlme-metadata))
+4. If a single file will exists (i.e. for a large CSV file) set the `only` setting. Example: `only: !ruby/regexp '/egyptian-20170820.csv/'`
+5. Assign the traject configuration file: `traject_file: my_new_source_config`
+6. Assign any appropriate properties. Example:
+```
+`agg_provider`: [NAME OF DATA PROVIDER]
+`inst_id`: short id prefix [i.e. `my_source`]
+```
+
+#### Example configuration
+
+```
+stanford_mods:
+      directory: maps/records/stanford
+      traject_file: stanford_mods_config
+      properties:
+        agg_provider: Stanford Libraries
+        agg_data_provider: Stanford Libraries
+        inst_id: stanford
+```
+
+### Data mapping <a name="mapping" />
+
+Review the [application profile](application_profile.md) to determine available and required fields.
+These are the field names that will be used in your configuration mapping.
+
+### Test data mapping <a name="test" />
+
+Add a `my_new_source_spec.rb` file to `spec/import`.
+
+#### Required test structure
+
+```
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Transforming My New Data Source file' do
+  let(:indexer) { Pipeline.for('[MY_ID_USED_IN_SETTINGS]').indexer(HarvestedResource.new(original_filename: fixture_file_path)) }
+  let(:fixture_file_path) { File.join(fixture_path, 'csv/[my_new_fixture file]') }
+  let(:data) { File.open(fixture_file_path).read }
+  let(:exhibit) { create(:exhibit) }
+  let(:slug) { exhibit.slug }
+
+  before do
+    indexer.settings['exhibit_slug'] = slug
+    allow(CreateResourceJob).to receive(:perform_later)
+  end
+
+  it 'does the transform' do
+    indexer.process(data)
+    expect(CreateResourceJob).to have_received(:perform_later) do |_id, _two, json|
+      dlme = JSON.parse(json)
+      expect(dlme['agg_provider']).to eq 'EXPECTED VALUE FROM FIXTURE'
+      ..repeat for all fields to test..
+    end
+  end
+end
+```
+
+### Mapping configuration <a name="config" />
```

Comment:
Should we link to the upstream traject docs?

Reply:
For sure. I can add that.

Reply:
Added link to traject project.